### PR TITLE
Host Release Prep PR v4.35.0

### PR DESCRIPTION
**Pull request by v-smanchem**
Bumping up the minor version to 4.35.0 and updating the release notes.

### Pull request checklist

**IMPORTANT**: Currently, changes must be backported to the `in-proc` branch to be included in Core Tools and non-Flex deployments.

* [ ] Backporting to the `in-proc` branch is not required
    * Otherwise: Link to backporting PR
* [x] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
    * [ ] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] My changes **do not** require diagnostic events changes
    * Otherwise: I have added/updated all related diagnostic events and their documentation (Documentation issue linked to PR)
* [ ] I have added all required tests (Unit tests, E2E tests)

### Additional information

Additional PR information

### CODE REVIEW

File: release_notes.md [hunk 1]
```diff
@@ -5,12 +5,6 @@
...
 - Update Python Worker Version to [4.29.0](https://github.com/Azure/azure-functions-python-worker/releases/tag/4.28.1)
-- Fixed an issue causing sporadic HTTP request failures when worker listeners were not fully initialized on first request #9954
-- Update Node.js Worker Version to [3.10.0](https://github.com/Azure/azure-functions-nodejs-worker/releases/tag/v3.10.0) (#9999)
-- Update PowerShell worker 7.2 to [4.0.3220](https://github.com/Azure/azure-functions-powershell-worker/releases/tag/v4.0.3220)
-- Update PowerShell worker 7.4 to [4.0.3219](https://github.com/Azure/azure-functions-powershell-worker/releases/tag/v4.0.3219)
-- Update Azure.Identity to 1.11.0 (#10002)
-- Fixed an issue leading to a race when invocation responses returned prior to HTTP requests being sent in proxied scenarios.
 - Language worker channels will not be started during placeholder mode if we are in-process (#10161)
 - Ordered invocations are now the default (#10201)
 - Skip worker description if none of the profile conditions are met (#9932)
```
Comment (kshyju) [inline release_notes.md:10]: Should we update the release_notes to reflect https://github.com/Azure/azure-functions-host/pull/10263?

Comment (v-imohammad) [reply release_notes.md:10]: We can proceed with updating the release notes, as we have already made updates in accordance with the process. @kshyju

**Review by kshyju — APPROVED** [review]

File: release_notes.md (hunk 1 above)
Comment (fabiocav) [reply release_notes.md:10]: Can you clarify the above? Do we have the notes reflecting what comes from the release notes file?

Comment (fabiocav) [reply release_notes.md:10]: To be clear, this is the state: https://github.com/Azure/azure-functions-host/blob/dev/release_notes.md

Comment (v-imohammad) [reply release_notes.md:10]: https://github.com/Azure/azure-functions-host/blob/in-proc/release_notes.md as per the Inproc branch
could you please review and approve

**Pull request closed without merging**